_posts/2018-03-09-starter-kit.md

```diff
@@ -0,0 +1,176 @@
+---
+title: Starter Kit - the turn-key template for your own pages
+author: pitti
+date: 2018-03-09
+category: tutorial
+tags: cockpit starter-kit
+slug: cockpit-starter-kit
+comments: true
+---
+
+## The bare minimum
+
+[Cockpit's API](http://cockpit-project.org/guide/latest/development.html) makes it easy to create your own pages (or
+"plugins" if you will) that appear in Cockpit's menu and interact with your system in any way you like. Our pet
+example is the [Pinger](https://github.com/cockpit-project/cockpit/tree/master/examples/pinger) which is just the bare
+minimum: a [HTML file](https://github.com/cockpit-project/cockpit/blob/master/examples/pinger/ping.html) with a form to
+enter an IP and a small piece of JavaScript to call the `ping` Linux command  through Cockpit
+[spawn()](http://cockpit-project.org/guide/latest/cockpit-spawn.html) and capture its output; and a
+[manifest file](https://github.com/cockpit-project/cockpit/blob/master/examples/pinger/manifest.json) which tells
+cockpit how to add it to the menu and where the entry point is.
+
+There is a rather old [blog post](http://cockpit-project.org/blog/creating-plugins-for-the-cockpit-user-interface.html)
+which explains the Pinger example in detail. Cockpit changed its visual design quite dramatically since then, and Pinger
+does not use jQuery any more, but aside from these details that post is still applicable.
+
+## Requirements for real projects
+
+Pinger is great for explaining and understanding the gist of how Cockpit works.  But an actual production-ready requires
+a lot more:
+
+ * Separation of HTML, CSS, and JavaScript code: This avoids the `unsafe-inline`
+   [Content-Security-Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP).  We *strongly* recommend this for
+   third-party pages, and absolutely require this for Cockpit's own pages.
```

Comment:
I sent https://github.com/cockpit-project/cockpit/pull/8806 to improve our pinger (and add a test). I adjusted the blog post accordingly.

I started a TODO list in the description to keep track of the improvements that should land before announcing this, so that the blog post matches reality.